### DISCUSSION TITLE
Creating PR's to upstream

### DISCUSSION THREAD

**Discussion by bjorn-out** [community/community#3408] [Codespaces] [answered]
It would be really great to be able to open PR's to an upstream repository entirely from a Codespace. Currently, I get an error once I select "Submit" after running `gh pr create`. It works if I continue in a browser, but that adds another step.

This would be especially great as you can auto-fork repo's right from VSCode. Being able to create PR's from the terminal as well would make it possible to very quickly make a change and create a PR for any repository.

The error I currently get is:
`pull request create failed: GraphQL error: Resource not accessible by integration`

**Comment by 2percentsilk** [comment, marked as answer]
Hi @bjorn-out,

It looks like we recently deployed a fix that should address the authorization problems you're encountering here. Could you try creating a new codespace and trying this workflow again? If it's still reproducing, we can take another step to investigate.

**Reply by bjorn-out** [reply]
Hi!

It's still reproducing, with the same error:
`pull request create failed: GraphQL error: Resource not accessible by integration`

**Reply by 2percentsilk** [reply]
Hi @bjorn-out, 

Unfortunately, I misspoke earlier - this is a known issue the team is aware of and working on. 
By default, we scope permissions to the repository your codespace is built off-of so a side effect is that you do not have permissions to create a PR against the upstream repository from within the codespace.

However, I have a workaround for you in the meantime 😄
1. Create a [Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token#creating-a-token) with the following permissions:
  - `public_repo`
  - `read:discussion`
  - `read:org`
  - Depending on the permissions of the upstream repo, you may need additional permissions but the resulting terminal error should let you know which ones! 
2. Set the environment variable `GITHUB_TOKEN` to equal your PAT.
  - You can do this by running `export GITHUB_TOKEN='PAT'` in your terminal.

Let me know how that goes and rest assured, we're working on streamlining this and hope to share it with you soon 🎉

**Comment by PranamBhat** [comment]
Hi @bjorn-out 

Check out here:

[https://github.com/actions/labeler/issues/12](https://github.com/actions/labeler/issues/12)

[https://github.com/alex-page/add-new-issue-project/issues/5](https://github.com/alex-page/add-new-issue-project/issues/5)

[https://github.com/actions/virtual-environments/issues/1609] (https://github.com/actions/virtual-environments/issues/1609)

[https://github.community/t/resource-not-accessible-by-integration-when-requesting-github-pages-build-via-rest-api/13567](https://github.community/t/resource-not-accessible-by-integration-when-requesting-github-pages-build-via-rest-api/13567)
